### PR TITLE
GatherableSystem/Component

### PR DESCRIPTION
**Pull request by HoofedEar**
## About the PR 
This is a functionally similar (but more customizable) replacement for MineableSystem and it's accompanying PickaxeComponent.

**Why tho?**
Currently, MineableSystem is what handles using a pickaxe on an asteroid rock to mine ores. It is mining specific, and not very flexible. Right now, it's limited to only allowing interaction with stuff that has the "Pickaxe" component (which means you can't limit or whitelist what tools you want to allow) thus limiting its use elsewhere. So instead of making a replica system for something like fishing or woodcutting or anything else, GatherableSystem can handle such use cases.

**What makes it useful?**

- Tag whitelist for tools accepted by the Gatherable entity (check if you are using the right tool)
- The loot datafield allows you to define a different EntityLootTable ID for each tag (or you can use "All" instead of the name of the tag to use the same loot table for all tags)
image
- GatheringTool replaces Pickaxe as a component, and is functionally identical. But now, you must be aware of the tags the item has as it is necessary for Gatherable to recognize it.

**Changelog**

N/A - No gameplay changes

### CODE REVIEW

File: Resources/Prototypes/tags.yml [hunk 1]
```diff
@@ -245,6 +245,9 @@
 
 - type: Tag
   id: Plastic
+  
+- type: Tag
+  id: Pickaxe
```
Comment (metalgearsloth) [inline Resources/Prototypes/tags.yml:250]: Alphabetical please.

File: Content.Server/Gatherable/Components/GatherableComponent.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    ///        - Tag1
+    ///        - Tag2
+    ///     mappedLoot:
+    ///       Tag1:
+    ///         - id: EntityID1
+    ///       FishingPole:
+    ///         - id: EntityID2
+    /// </summary>
+    [DataField("loot")] 
+    public Dictionary<string, List<EntitySpawnEntry>>? MappedLoot = new();
```
Comment (metalgearsloth) [inline Content.Server/Gatherable/Components/GatherableComponent.cs:34]: We have an entityloottable prototype which is probably more suited for this.

File: Resources/Prototypes/tags.yml (hunk 1 above)
Comment (HoofedEar) [reply Resources/Prototypes/tags.yml]: Plastic was out of order, I put it in the proper spot, so it's now in proper alphabetical order.

File: Content.Server/Gatherable/Components/GatherableComponent.cs (hunk 2 above)
Comment (HoofedEar) [reply Content.Server/Gatherable/Components/GatherableComponent.cs]: Updated to use EntityLootTable (epico)

**Review by metalgearsloth — APPROVED** [review]

**Pull request merged**